cmd/metrics_exporter/metrics_exporter.go

```diff
@@ -0,0 +1,356 @@
+// Licensed to Yugabyte, Inc. under one or more contributor license
+// agreements. See the NOTICE file distributed with this work for
+// additional information regarding copyright ownership. Yugabyte
+// licenses this file to you under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package metrics_exporter
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/yugabyte/ybm-cli/cmd/util"
+	ybmAuthClient "github.com/yugabyte/ybm-cli/internal/client"
+	"github.com/yugabyte/ybm-cli/internal/formatter"
+	ybmclient "github.com/yugabyte/yugabytedb-managed-go-client-internal"
+)
+
+var MetricsExporterCmd = &cobra.Command{
+	Use:   "metrics-exporter",
+	Short: "Manage Metrics Exporter",
+	Long:  "Manage Metrics Exporter",
+	Run: func(cmd *cobra.Command, args []string) {
+		cmd.Help()
+	},
+}
+
+var createMetricsExporterCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Create Metrics Exporter Config",
+	Long:  "Create Metrics Exporter Config",
+	Run: func(cmd *cobra.Command, args []string) {
+
+		metricsExporterName, _ := cmd.Flags().GetString("name")
+		metricsSinkType, _ := cmd.Flags().GetString("type")
+		datadogSpecString, _ := cmd.Flags().GetStringToString("datadog-spec")
+		apiKey := datadogSpecString["api-key"]
+		site := datadogSpecString["site"]
+
+		if strings.ToLower(metricsSinkType) == "datadog" {
```

Comment:
Please refactor a bit - the above DD related items should only be in the if branch.

Reply:
They aren't actually DD specific, ApiKey and Site will be used in others as well